### PR TITLE
rsx: Implement signed normalized texture formats

### LINKED ISSUE #4025: Unreal Engine 3 games issues (common)
I suggest to write here about all found bugs and issues on Unreal Engine 3 games since a lot of them seems require fixing for now.

For Silent Hill Downpour:
1. No dynamic shadows from player and objects.
2. Static world lighting computes incorrectly: most of areas look overbrighten and overbloomed.
3. Bad model lighting: some models looks completely white or too dark,
4. Intro cutscene's video freezes and doesn't exit after played, so the following game progress is impossible.

The same issues (except 4) noticed on Saw: The videogame.

### LINKED ISSUE #6525: Drakengard 3 Branch D Final boss Seam Shading
The shading makes seams on the models very obvious. It also makes the shadows much darker than they were originally.

image
in comparison to
image

### PR DESCRIPTION
**Pull request by kd-11**
This was already partially supported via EXP option in the shader opcode, but format decoding was disabled. Discovered in some UE3 games which use _SNORM variants on PC but _UNORM on rpcs3 because channel signing was disabled.

Fixes unreal engine 3 lighting bugs among other things.

Fixes #6525
Fixes #4025

### CODE REVIEW

**Comment by AniLeo** [comment]
Fixes lighting on **Drakengard 3**

**Master**
Screenshot from 2020-06-10 23-41-39

**PR**
Screenshot from 2020-06-10 23-37-55

**Comment by JohnHolmesII** [comment]
Graphics look better on ME1, but Shepard's eyes are missing now for some reason.

<details>
<summary>Master</summary>

Master

</details>

<details>
<summary>PR</summary>

PR
</details>

ME2
<details>
<summary>Master</summary>

Master

</details>

<details>
<summary>PR</summary>

PR
</details>

**Comment by Emulator-Team-2** [comment]
Fixes lighting in (Spider-Man: Edge of Time, Asura's Wrath, Resistance: Fall of Man).

**Before/After Comparisons (Click to Expand)**

<details> 
<summary>Spider-Man: Edge of Time</summary>

**Before:**

**After:**


</details>

<details> 
<summary>Asura's Wrath</summary>

**Before:**

**After:**


</details>

<details> 
<summary>Resistance: Fall of Man</summary>

**Before:**

**After:**

</details>

**Comment by MarioSonic2987** [comment]
Fixes broken shading in BioShock Infinite.

**Before**:
image

**After**:
image

**Comment by digitaldude555** [comment]
Resistance 2 before and after
Screenshot (2370)
Screenshot (2380)

**Comment by DefaltBR** [comment]
Fixes all UE3 common issues in #4025 (except an issue #4641 mentioned there, which is game related - SH Downpour)

**Comment by DefaltBR** [comment]
Fixes all UE3 common issues in #4025 (only thing left is probably a game-related issue #4641 - "binoculars" not working in SH Downpour)

**Comment by zombie167** [comment]
Fixes broken shading in Ninja Gaiden Sigma 2 cutscenes

**Master**

u2

**PR**

f2


**Master**

u1


**PR**

f1

**Comment by kd-11** [comment]
I've had to restructure the implementation a bit to fix that regression in mass effect due to stacked modifiers. Please retest.

File: rpcs3/Emu/RSX/RSXTexture.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ namespace rsx
 		rsx::texture_minify_filter  min_filter() const;
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/RSXTexture.h:69]: double space on 4 functions

Comment (Megamouse) [reply rpcs3/Emu/RSX/RSXTexture.h:69]: also:
max_aniso

Comment (Megamouse) [reply rpcs3/Emu/RSX/RSXTexture.h:69]: looks like this is some old formatting, looking at even more functions above

Comment (kd-11) [reply rpcs3/Emu/RSX/RSXTexture.h:69]: yes, its old formatting

**Comment by kd-11** [comment]
Moved to WIP, its broken again.

**Comment by kd-11** [comment]
Rolled back last commit which broke everything, I need a different implementation to make mass effect work correctly.

**Comment by kd-11** [comment]
New implementation, seems to work correctly for me. Please confirm everything is still fixed if you can.

**Comment by weter11** [comment]
Drakengard 3. Master: building from left looks correct, shadows from dragon & right building not.
Снимок экрана_2020-06-11_22-13-21
PR: correct shadows from dragon & right building, left building became too bright.
Снимок экрана_2020-06-11_23-29-03
PS3.
Снимок экрана_2020-06-11_21-17-121
Shader cache was always deleted before running.

**Comment by kd-11** [comment]
@weter11 Did you compare the first and second commit? Or which commit is your test based on? There have been 2 versions of this PR so far.

**Comment by weter11** [comment]
8f9e23df

**Comment by DefaltBR** [comment]
Everything seems to be in order with latest Windows PR (build 8f9e23d)! Retested a few UE3 games (Vulkan) and it works like a charm for fixing lighting/shadowing/shading problems! (Only a few minor game specific bugs are left, which were already present before PR 8398 and are not related to shadowing/lighting!)

Games that I've tested and played for extended times with the latest PR build (and 1st PR build) and can confirm that are still fixed:

- LOLLIPOP CHAINSAW [BLUS30917]
- SAW [BLES00676]
- SAW II [BLES01050]
- Shadows of the Damned [BLUS30653]
- Silent Hill Downpour [BLUS30565]
- The Bourne Conspiracy [BLES00250]

**Comment by MarioSonic2987** [comment]
All games I've reported in this PR are still improved. I didn't notice any regressions.

**Comment by i2puni** [comment]
kd-11 great!!

DMC  (Devil May Cry)
shading and lighting Fixes

**Comment by Parthorisian** [comment]
@i2puni
ye so epic unexpected fixes for for things being glitched during last 2years, nice

**Pull request merged**